driver/src/main/com/mongodb/client/MongoDatabase.java

```diff
@@ -191,4 +191,19 @@
      */
     void createCollection(String collectionName, CreateCollectionOptions createCollectionOptions);
 
+    /**
```

Comment:
I don't think that MongoDatabase should be a factory for GridFSBucket instances.  GridFS is an application on top of MongoDB, and therefore GridFSBucket should depend on MongoDatabase, not the other way around.  

The advantage of this approach is that we don't have to expose a concrete implementation of GridFSBucket, but I still think it's a mistake.

How about 

public class GridFSBuckets {
     public static GridFSBucket create(MongoDatabase);
}